url/tools/IdnaTestV2-parser.py

```diff
@@ -0,0 +1,147 @@
+
+    for line in lines:
+        # Remove newlines
+        line = line.rstrip()
+
+        # Remove lines that are comments or empty
+        if line.startswith("#") or line == "":
+            continue
+
+        # Remove escapes (doesn't handle \x{XXXX} but those do not appear in the source)
+        line = remove_escapes(line)
+
+        # Normalize columns
+        #
+        # Since we are only interested in ToASCII and enforce Transitional_Processing=false we care
+        # about the following columns:
+        #
+        # * Column 1: source
+        # * Column 4: toAsciiN
+        # * Column 5: toAsciiNStatus
+        columns = [column.strip() for column in line.split(";")]
+
+        # Column 1
+        column_source = columns[0]
+
+        if exclude_ipv4_like:
+            if ends_in_a_number(column_source):
+                continue
+
+        if exclude_std3_non_ascii:
+            # Notably this does not deal with them appearing in xn-- inputs.
+            if re.search(r"\u2260|\u226E|\u226F", column_source):
+                continue
+
+        # Column 4 (if empty, use Column 2; if empty again, use Column 1)
+        column_to_ascii = columns[3]
+        if column_to_ascii == "":
+            column_to_ascii = columns[1]
+            if column_to_ascii == "":
+                column_to_ascii = column_source
+
+        # Column 5 (if empty, use Column 3)
+        column_status = columns[4]
+        if column_status == "":
+            column_status = columns[2]
+        # Convert to list
+        if column_status == "":
+            column_status = []
+        else:
+            assert column_status.startswith("[")
+            column_status = [status.strip() for status in column_status[1:-1].split(",")]
+
+        # Side quest time.
+        for status in column_status:
+            if status not in unique_statuses:
+                unique_statuses.append(status)
+
+        # The URL Standard has
+        #
+        # * UseSTD3ASCIIRules=false; however there are no tests marked U1 (some should be though)
+        # * CheckHyphens=false; thus ignore V2, V3?
+        # * VerifyDnsLength=false; thus ignore A4_1 and A4_2
+        comment = ""
+        for ignored_status in ["A4_1", "A4_2", "U1", "V2", "V3"]:
+            if ignored_status in column_status:
+                column_status.remove(ignored_status)
+                comment += ignored_status + " (ignored); "
+        for status in column_status:
+            comment += status + "; "
+        if comment != "":
+            comment = comment.strip()[:-1]
+
+        if len(column_status) > 0:
+            column_to_ascii = None
+
+        test = { "input": column_source, "output": column_to_ascii }
+        if comment != "":
+            test["comment"] = comment
+        output.append(test)
+
+    unique_statuses.sort()
+    return { "tests": output, "unique_statuses": unique_statuses }
+
+def to_json(data):
+    handle = open("../resources/IdnaTestV2.json", "w")
```

Comment:
Do we want to pass the path via command line? This seems relative to the CWD. Or we can use `os.path.join(os.path.dirname(__file__), "../resources/IdnaTestV2.json")` instead

Reply:
Interesting, that's a lot of boilerplate, but I guess it makes it slightly more robust. I'd have to make a similar change for the `IdnaTestV2.txt` file presumably.